src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_ILockStatement.cs

```diff
@@ -1025,5 +1187,840 @@ void M(P input, bool b)
+                                  Value:
+                                    IParameterReferenceOperation: l2 (OperationKind.ParameterReference, Type: System.Threading.Lock) (Syntax: 'l2')
+                            Next (Regular) Block[B4]
+                        Block[B4] - Block
+                            Predecessors: [B2] [B3]
+                            Statements (1)
+                                IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'b ? l1 : l2')
+                                  Value:
+                                    IInvocationOperation ( System.Threading.Lock.Scope System.Threading.Lock.EnterLockScope()) (OperationKind.Invocation, Type: System.Threading.Lock.Scope, IsImplicit) (Syntax: 'b ? l1 : l2')
+                                      Instance Receiver:
+                                        IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.Threading.Lock, IsImplicit) (Syntax: 'b ? l1 : l2')
+                                      Arguments(0)
+                            Next (Regular) Block[B5]
+                                Leaving: {R2}
+                                Entering: {R3} {R4}
+                    }
+                    .try {R3, R4}
+                    {
+                        Block[B5] - Block
+                            Predecessors: [B4]
+                            Statements (0)
+                            Next (Regular) Block[B7]
+                                Finalizing: {R5}
+                                Leaving: {R4} {R3} {R1}
+                    }
+                    .finally {R5}
+                    {
+                        Block[B6] - Block
+                            Predecessors (0)
+                            Statements (1)
+                                IInvocationOperation ( void System.Threading.Lock.Scope.Dispose()) (OperationKind.Invocation, Type: System.Void, IsImplicit) (Syntax: 'b ? l1 : l2')
+                                  Instance Receiver:
+                                    IFlowCaptureReferenceOperation: 1 (OperationKind.FlowCaptureReference, Type: System.Threading.Lock.Scope, IsImplicit) (Syntax: 'b ? l1 : l2')
+                                  Arguments(0)
+                            Next (StructuredExceptionHandling) Block[null]
+                    }
+                }
+                Block[B7] - Exit
+                    Predecessors: [B5]
+                    Statements (0)
+                """;
+
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>([source, LockTypeDefinition], expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [Fact, CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        public void LockFlow_LockObject_Conditional_Object()
```

Comment:
We should also have tests with conditional flow inside the body of the lock itself.